server/util/clickhouse/clickhouse.go

```diff
@@ -302,23 +303,63 @@ func ToInvocationFromPrimaryDB(ti *tables.Invocation) *Invocation {
 	}
 }
 
+func ToExecutionFromInternalProto(in *iepb.Execution, inv *tables.Invocation) *Execution {
```

Comment:
nit: can this be private?

also consider `buildExecution` since it builds it from both the internal proto and the DB invocation -- or split into separate funcs like `fillExecutionFromInternalProto`, `fillExecutionFromInvocation`.